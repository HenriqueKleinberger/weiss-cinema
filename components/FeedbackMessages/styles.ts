import styled from 'styled-components';

export const Container = styled.div`
  color: var(--secondary);
  display: flex;
  flex-direction: column;
  gap: 2rem;
  align-items: center;
  justify-content: center;
  fill: currentColor;
  width: 100%;
`;
