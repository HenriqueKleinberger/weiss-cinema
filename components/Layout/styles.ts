import styled from 'styled-components';

interface ITab {
  active: boolean;
}

export const Container = styled.div`
  padding: 0.5rem 2rem;
  min-height: 100vh;
  background-color: var(--background);
`;
export const Logo = styled.div`
  color: var(--primary);
  fill: currentColor;
  align-items: center;
  display: flex;
  font-weight: 500;
  letter-spacing: 1px;
  gap: 1rem;
`;

export const Header = styled.header`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;

  @media (max-width: 600px) {
    grid-template-columns: 1fr 1fr;
  }
`;

export const Navigation = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-around;
`;

export const Tab = styled.a<ITab>`
  text-decoration: ${(props) => (props.active ? 'underline' : '')};
  font-weight: ${(props) => (props.active ? '800' : '400')};
  text-transform: uppercase;
  color: var(--white);
  cursor: pointer;
`;
