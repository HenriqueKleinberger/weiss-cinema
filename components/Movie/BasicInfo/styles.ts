import styled from 'styled-components';

export const Data = styled.div`
  color: var(--secondary);
  text-align: center;
  font-weight: 600;
  margin-top: 30%;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: wrap;
`;

export const Hovered = styled.div`
  z-index: 1;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  cursor: pointer;
`;

export const Wishlist = styled.div`
  margin: 7%;
`;
