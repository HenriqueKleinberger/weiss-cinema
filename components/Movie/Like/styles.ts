import styled from 'styled-components';

interface IBtn {
  active: boolean;
}

export const Button = styled.button<IBtn>`
  background: none;
  border: none;
  padding: 0;
  cursor: pointer;
  outline: red;
  stroke: var(--primary);
  stroke-width: 25px;
  fill: ${(props) => (props.active ? 'var(--primary)' : '#000000')};
  z-index: 1;
`;
