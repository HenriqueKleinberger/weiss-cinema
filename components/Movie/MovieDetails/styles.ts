import styled from 'styled-components';
import { keyframesFullView, shimmerAnimation } from '../../../layout/utils';

export const PosterLoading = styled.div`
  height: 100%;
  width: 100%;
  animation: ${keyframesFullView} 0.5s forwards
    cubic-bezier(0.25, 0.46, 0.45, 0.94);
  ${shimmerAnimation}
`;

export const InfoLoading = styled.div`
  height: 1rem;
  margin: 0.5rem;
  ${shimmerAnimation}
`;

export const Container = styled.div`
  position: fixed;
  inset: 0; /* inset sets all 4 values (top right bottom left) much like how we set padding, margin etc., */
  background-color: rgba(0, 0, 0, 0.7);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  transition: all 0.3s ease-in-out;
  overflow: hidden;
  z-index: 2;
  padding: 2.3rem 1.2rem 1.2rem;
`;

export const Data = styled.aside`
  height: 65%;
  width: 50%;
  background-color: var(--background);
  color: var(--secondary);
  display: grid;
  grid-template-columns: 1.5fr 2fr;
  overflow: auto;

  @media (max-width: 992px) {
    height: 75%;
    width: 65%;
  }
  @media (max-width: 778px) {
    height: 85%;
    width: 80%;
  }
`;

export const ErrorData = styled.aside`
  height: 60%;
  width: 50%;
  background-color: var(--background);
  color: var(--secondary);
  display: flex;
  justify-content: center;
  align-content: center;
  flex-direction: column;
  text-align: center;
`;

export const Details = styled.section``;
export const Title = styled.h1`
  margin: 0 0.2rem 0 0;
`;
export const Year = styled.p`
  margin: 0;
  font-size: 0.7rem;
`;
export const MainDetails = styled.div`
  color: var(--white);
  margin: 0.5rem;
`;
export const OtherDetails = styled.section`
  margin: 0.5rem;
`;

export const Row = styled.div`
  display: flex;
`;
