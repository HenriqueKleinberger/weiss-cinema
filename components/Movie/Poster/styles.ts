import styled from 'styled-components';

export const Frame = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
`;
