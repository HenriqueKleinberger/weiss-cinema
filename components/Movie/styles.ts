import styled from 'styled-components';

interface IBox {
  isHovered: boolean;
}

export const Container = styled.div`
  position: relative;
  width: 200px;
  height: 300px;
`;
export const Box = styled.div<IBox>`
  width: 100%;
  height: 100%;
  position: absolute;
  opacity: ${(props) => (props.isHovered ? '0.1' : '1')};
  top: 0;
  left: 0;
`;
