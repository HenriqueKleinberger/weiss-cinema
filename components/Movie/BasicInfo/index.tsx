import { IMovie } from '../../../types';
import { Data, Hovered, Wishlist } from './styles';
import Like from '../Like';

interface IProps {
  movie: IMovie;
  removeMovieFromWishlist?: (movie: IMovie) => void;
}

const BasicInfo = ({ movie, removeMovieFromWishlist }: IProps) => {
  const getMovieTitle = () => {
    if (movie.title.length >= 90)
      return `${movie.title.slice(0, 90).toUpperCase()}...`;
    else return movie.title.toUpperCase();
  };

  return (
    <Hovered>
      <Wishlist>
        <Like movie={movie} removeMovieFromWishlist={removeMovieFromWishlist} />
      </Wishlist>
      <Data>
        <p>{getMovieTitle()}</p>
        <p>{movie.year}</p>
      </Data>
    </Hovered>
  );
};

export default BasicInfo;
