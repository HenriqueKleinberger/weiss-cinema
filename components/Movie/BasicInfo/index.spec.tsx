import React from 'react';
import { shallow } from 'enzyme';
import BasicInfo from './index';

import {
  isMovieOnLocalStorage,
  toggleMovieOnLocalStorage,
} from '../../../helpers/localStorageWishlist';
import { Like } from './styles';

jest.mock('../../../helpers/localStorageWishlist', () => ({
  toggleMovieOnWishlist: jest.fn(),
  isMovieOnWishlist: jest.fn(),
  getMoviesFromLocalStorage: jest.fn(),
}));

describe('BasicInfo', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot when movie is not on wishlist', () => {
    (isMovieOnLocalStorage as jest.Mock).mockReturnValue(false);
    const wrapper = shallow(
      <BasicInfo
        movie={{
          imdbID: '1234',
          poster: 'poster',
          title: 'Title',
          year: '2022',
        }}
      />
    );
    expect(wrapper.find(Like).at(0).props().active).toBe(false);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when movie is on wishlist', () => {
    (isMovieOnLocalStorage as jest.Mock).mockReturnValue(true);
    const wrapper = shallow(
      <BasicInfo
        movie={{
          imdbID: '1234',
          poster: 'poster',
          title: 'Title',
          year: '2022',
        }}
      />
    );
    expect(wrapper.find(Like).at(0).props().active).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('title needs to be shorten when is bigger than 90 characters', () => {
    (isMovieOnLocalStorage as jest.Mock).mockReturnValue(true);
    const wrapper = shallow(
      <BasicInfo
        movie={{
          imdbID: '1234',
          poster: 'poster',
          title:
            'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Nullam feugiat, turpis at pulvinar vulput',
          year: '2022',
        }}
      />
    );
    expect(wrapper.find('p').at(0).text()).toBe(
      'LOREM IPSUM DOLOR SIT AMET, CONSECTETUER ADIPISCING ELIT. NULLAM FEUGIAT, TURPIS AT PULVIN...'
    );
  });

  it('should call toggleMovieOnWishlist when click the like button', () => {
    const stopPropagation = jest.fn();
    const removeMovieFromWishlist = jest.fn();
    const wrapper = shallow(
      <BasicInfo
        movie={{
          imdbID: '1234',
          poster: 'poster',
          title: 'Title',
          year: '2022',
        }}
        removeMovieFromWishlist={removeMovieFromWishlist}
      />
    );
    wrapper.find(Like).at(0).simulate('click', { stopPropagation });

    expect(toggleMovieOnLocalStorage).toHaveBeenCalledTimes(1);
    expect(stopPropagation).toHaveBeenCalledTimes(1);
    expect(removeMovieFromWishlist).toHaveBeenCalledTimes(1);
  });
});
