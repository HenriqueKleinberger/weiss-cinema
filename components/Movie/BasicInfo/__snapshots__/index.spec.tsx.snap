// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <Like
      movie={
        Object {
          "imdbID": "1234",
          "poster": "poster",
          "title": "Title",
          "year": "2022",
        }
      }
    />
  </styled.div>
  <styled.div>
    <p>
      TITLE
    </p>
    <p>
      2022
    </p>
  </styled.div>
</styled.div>
`;
